worker/payingAction.js

```diff
@@ -0,0 +1,158 @@
+import { getPaymentFailureStatus, getPaymentOrNotSent } from '@/api/lnd'
+import { walletLogger } from '@/api/resolvers/wallet'
+import { formatMsats, formatSats, msatsToSats, toPositiveBigInt } from '@/lib/format'
+import { datePivot } from '@/lib/time'
+import { notifyWithdrawal } from '@/lib/webPush'
+import { Prisma } from '@prisma/client'
+
+async function transitionWithdrawal (jobName,
+  { withdrawalId, toStatus, transition, withdrawal, onUnexpectedError },
+  { models, lnd, boss }
+) {
+  console.group(`${jobName}: transitioning withdrawal ${withdrawalId} from null to ${toStatus}`)
+
+  let dbWithdrawal
+  try {
+    const currentDbWithdrawal = await models.withdrawl.findUnique({ where: { id: withdrawalId } })
+    console.log('withdrawal has status', currentDbWithdrawal.status)
+
+    if (currentDbWithdrawal.status) {
```

Comment:
initial state of withdrawals is a null status and there are not intermediate states, so if it's not null, it's in a terminal state